### PR TITLE
ES-618: Add Gradle version catalog implementation for corda-runtime-os

### PR DESCRIPTION
**Pull request by knguyen-r3**
**Description:**
As part improving version dependencies within Corda we have added [Gradle Version Catalog](https://docs.gradle.org/current/userguide/platforms.html) for corda-runtime-os. This will allow us to use the newly created central repository [corda-version-catalog](https://github.com/corda/corda-version-catalog) for maintaining and managing 3rd parties dependency that is use within corda-runtime-os.

**Changes:**
Added implementation of using Gradle Version Catalog for corda-runtime-os repo in build.gradle file
Replace all version reference occurrences which use gradle.properties to version catalog in build.gradle files.

**Testing**
Tested with Artifactory cordaVersionCatalog=5.0.0-alpha-1687973674215 created from this [PR](https://github.com/corda/corda-version-catalog/pull/4) in version catalog repo.

kotlinMetadataVersion=0.6.2
internalDockerVersion=1.+ (not used)
snakeyamlVersion=2.0
activationVersion=1.2.0
ariesDynamicFrameworkExtensionVersion=1.3.6
antlrVersion=2.7.7
asmVersion=9.4
avroVersion=1.11.1
commonsVersion = 1.7
caffeineVersion = 3.1.6
commonsLangVersion = 3.12.0
commonsTextVersion = 1.10.0
bouncycastleVersion=1.73
disruptorVersion=3.4.4
felixSecurityVersion=2.8.4
guavaVersion=32.1.1-jre
hibernateVersion=5.6.15.Final
hikariCpVersion=5.0.1
jacksonVersion=2.15.0
jaxbVersion = 2.3.1
kafkaClientVersion=3.4.0_1
kryoVersion = 5.5.0
kryoSerializersVersion = 0.45
kotlinCoroutinesVersion=1.6.4
liquibaseVersion = 4.19.0
beanutilsVersion=1.9.4
micrometerVersion=1.11.0
nettyVersion = 4.1.94.Final
networkntJsonSchemaVersion = 1.0.66
osgiCmVersion = 1.6.1
osgiNamespaceServiceVersion = 1.0.0
picocliVersion = 4.7.3
protonjVersion=0.34.1
reflectAsmVersion = 1.11.9
snappyVersion=0.4
typeSafeConfigVersion=1.4.2
jsonCanonicalizerVersion=1.1
braveVersion=5.15.0
zipkinVersion=2.16.3
osgiJdbcServiceVersion=1.1.0
paxJdbcVersion=1.5.3
assertjVersion=3.24.2
dom4jOsgiVersion = 2.1.3_1
hamcrestVersion=2.2
hsqldbVersion=2.7.1
jimfsVersion = 1.2
mockitoKotlinVersion=4.1.0
mockitoVersion=5.3.1
postgresDriverVersion=42.6.0
slingVersion=3.3.6
javalinVersion = 4.6.7
swaggerVersion = 2.2.9
nimbusVersion = 10.8
jcipAnnotationsVersion = 1.0_2
unirestVersion = 3.14.2
jettyVersion = 9.4.51.v20230217
pf4jVersion=3.9.0
systemLambdaVersion=1.2.1

**Documentation**
Rendered version: https://github.com/corda/platform-eng-design/blob/knguyen/ES-374/gradle_version_catalog/core/corda-5/corda-5.0/infrastructure/build/build-dependencies/corda5-version-catalog.md

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -59,6 +59,43 @@ plugins {
...
+            maven {
+                url = "$artifactoryContextUrl/$cordaUseCache"
+                name = "R3 Maven remote repositories"
+                authentication {
+                    basic(BasicAuthentication)
+                }
+                credentials {
+                    username = settings.ext.find('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
+                    password = settings.ext.find('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD')
+                }
```
Comment (yarivus) [inline settings.gradle:77]: can this closure/scope/block be defined outside the maven scope and thus not having to write it twice ?

Comment (yarivus) [reply settings.gradle:77]: Same applies for:
```
authentication {
    basic(BasicAuthentication)
}
```

**Review by yarivus — COMMENTED** [review]
see question/comment

**Review by yarivus — COMMENTED** [review]
My question still pending ;)

File: settings.gradle (hunk 1 above)
Comment (knguyen-r3) [reply settings.gradle:77]: Yeah I think it's should be possible. Will add that in.

**Review by yarivus — COMMENTED** [review]
still pending comment

**Review by yarivus — COMMENTED** [review]
still pending comment

File: settings.gradle (hunk 1 above)
Comment (ronanbrowne) [reply settings.gradle:77]: @knguyen-r3 / @yarivus The DSL above is scoped to the outer blocks, i.e maven  / repositories blocks  etc, you will not be able to extract this to a common helper method as it simply wont compile as the DSL wont exist out of the expected Closure, Gradle is also more particular around passing objects around so one cant just pass the Maven object to a support method and set it there.

repeated logic which is not scoped can be extracted, to a support method however such as

```
String getValue(String property, String envVar) {
    return settings.ext.find(property) ?: System.getenv(envVar)
}
```

and then 
```
                    username = getValue('cordaArtifactoryUsername','CORDA_ARTIFACTORY_USERNAME')
                    password = getValue('cordaArtifactoryPassword', CORDA_ARTIFACTORY_PASSWORD'
```

Comment (yarivus) [reply settings.gradle:77]: Thanks for clarifying @ronanbrowne

**Review by yarivus — COMMENTED** [review]
Checkpoint review LGTM

**Review by yarivus — COMMENTED** [review]
checkpoint review LGTM

**Pull request closed without merging**